docs/plugins/schedulers.rst

```diff
@@ -1,46 +1,791 @@
+        # at the top of our script.
+        def get_lines(self):
+            """Get the sbatch header lines."""
+
+            lines = super().get_lines()
+
+            # Here we just add directives in the slurm sbatch format,
+            # according to the test's configuration.
+            lines.append(
+                '#SBATCH --job-name "pav test #{s._test_id}"'
+                .format(s=self))
+            lines.append('#SBATCH -p {s._conf[partition]}'.format(s=self))
+            if self._conf.get('reservation') is not None:
+                lines.append('#SBATCH --reservation {s._conf[reservation]}'
+                             .format(s=self))
+            if self._conf.get('qos') is not None:
+                lines.append('#SBATCH --qos {s._conf[qos]}'.format(s=self))
+            if self._conf.get('account') is not None:
+                lines.append('#SBATCH --account {s._conf[account]}'.format(s=self))
+
+You'll also need to override the ``_get_kickoff_script_header()`` method of
+your scheduler plugin to return an instance of your custom header class for use
+in the kickoff script.
+
+.. code-block:: python
+
+    def _get_kickoff_script_header(self, test):
+        """Get the kickoff header. Most of the work here """
+
+        sched_config = test.config[self.name]
+
+        # For the slurm scheduler, we store our node info under 'nodes'.
+        nodes = self.get_data()['nodes']
+
+        return SbatchHeader(sched_config,
+                            # This is where we handle our node filtering and
+                            # get a pre-formatted node range.
+                            self._get_node_range(sched_config, nodes.values()),
+                            test.id,
+                            self.get_vars(test))
+
+
+Scheduling a Test
+~~~~~~~~~~~~~~~~~
+
+The ``_schedule()`` method of your scheduler class is responsible for handing
+control of each test run to the scheduler and returning a job id for that run.
+
+Typically this involves running one or more shell commands to tell your
+scheduler to enqueue a command or script. This is typically done with the
+``subprocess`` module. Since Pavilion support Python 3.5+, you can use the
+new(ish) ``subprocess.run()`` function, though ``subprocess.Popen()`` may be
+more appropriate.
+
+
+Composing Commands
+^^^^^^^^^^^^^^^^^^
+
+You should compose your commands as a list. (Try to avoid the
+'shell=True' string based method. It tends to be error prone). Full paths
+to commands can be found with the distutils module.
+
+.. code-block:: python
+
+    import distutils.spawn
+
+    srun = distutils.spawn.find_executable('srun')
+    if srun is None:
+        raise SchedulerError
+
+    my_cmd = [srun]
+
+    # Building your commands with a list is simple and flexible.
+    if redirect_output:
+        my_cmd.extend(['-o', outfile])
+
+    mycmd.append('--partition=strange')
+
+    subprocess.run(my_cmd)
+
+To find commands on a system, the
```

Comment:
You've wandered off again.  I think this is covered at the top of this subsection.